integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/AwsGlueJobNameProvider.java

```diff
@@ -0,0 +1,91 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.lifecycle;
```

Comment:
Any suggestion where to put the providers? The package `lifecycle` seems a little bit off.

Reply:
`spark.agent.job.providers` ?